### PR TITLE
Multitenancy v2 changes for prompt studio apps - Prompt studio core, Registry

### PR DESCRIPTION
**Pull request by harini-venkataraman**
## What
Multitenancy v2 changes for Prompt studio, Prompt studio core and Prompt studio registry.

...

## Why
Performance improvements.
...

## How

...

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)
No, the code is not referred. 
...

## Database Migrations

- 

## Env Config

- 

## Relevant Docs

-

## Related Issues or PRs
https://github.com/Zipstack/unstract/pull/555/
https://github.com/Zipstack/unstract/pull/503/
https://github.com/Zipstack/unstract/pull/391
https://github.com/Zipstack/unstract/pull/564

-

## Dependencies Versions

-

## Notes on Testing
Untested code. Need to add migration files and test with migration.
-

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py [hunk 1]
```diff
@@ -12,10 +12,17 @@
 from django.conf import settings
 from django.db.models.manager import BaseManager
 from file_management.file_management_helper import FileManagerHelper
-from prompt_studio.prompt_profile_manager_v2.models import ProfileManager
-from prompt_studio.prompt_profile_manager_v2.profile_manager_helper import (
+from prompt_studio.modifier_loader import ModifierConfig
+from prompt_studio.modifier_loader import load_plugins as load_modifier_plugins
+from prompt_studio.prompt_profile_manager.models import ProfileManager
```
Comment (muhammad-ali-e) [inline backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py:17]: We need it from V2 instaed of V1

File: backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py [hunk 2]
```diff
@@ -29,13 +36,12 @@
...
     NoPromptsFound,
+    OperationNotSupported,
     PermissionError,
     ToolNotValid,
 )
-from prompt_studio.prompt_studio_core_v2.models import CustomTool
-from prompt_studio.prompt_studio_core_v2.prompt_ide_base_tool import PromptIdeBaseTool
-from prompt_studio.prompt_studio_document_manager_v2.models import DocumentManager
-from prompt_studio.prompt_studio_index_manager_v2.prompt_studio_index_helper import (  # noqa: E501
+from prompt_studio.prompt_studio_document_manager.models import DocumentManager
```
Comment (muhammad-ali-e) [inline backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py:43]: models should import from V2

File: backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py [hunk 3]
```diff
@@ -29,13 +36,12 @@
...
+from prompt_studio.prompt_studio_index_manager.prompt_studio_index_helper import (  # noqa: E501
```
Comment (muhammad-ali-e) [inline backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py:44]: Why it changed to V1?

**Review by muhammad-ali-e — CHANGES_REQUESTED** [review]

File: backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py (hunk 1 above)
Comment (muhammad-ali-e) [reply backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py]: updated

File: backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py (hunk 2 above)
Comment (muhammad-ali-e) [reply backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py]: updated

File: backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py (hunk 3 above)
Comment (muhammad-ali-e) [reply backend/prompt_studio/prompt_studio_core_v2/prompt_studio_helper.py]: updated

**Comment by hari-kuriakose** [comment]
Duplication will be handled when we remove v1 later.

File: backend/prompt_studio/prompt_studio_v2/models.py [hunk 4]
```diff
@@ -45,8 +46,8 @@ class Mode(models.TextChoices):
     )
     tool_id = models.ForeignKey(
         CustomTool,
-        on_delete=models.CASCADE,
-        related_name="tool_studio_prompts",
+        on_delete=models.SET_NULL,
```
Comment (hari-kuriakose) [inline backend/prompt_studio/prompt_studio_v2/models.py:49]: @harini-venkataraman Not sure if this is intended.

**Review by hari-kuriakose — COMMENTED** [review]
@harini-venkataraman Models look good overall.

Added a comment though.

**Pull request merged**